dipy/reconst/ivim.py

```diff
@@ -0,0 +1,585 @@
+                                    bounds=bounds,
+                                    ftol=ftol,
+                                    xtol=xtol,
+                                    gtol=gtol,
+                                    max_nfev=maxfev,
+                                    args=(self.gtab, data),
+                                    x_scale=self.x_scale)
+                ivim_params = res.x
+                return ivim_params
+            except:
+                return x0
+
+
+class IvimFit(object):
+
+    def __init__(self, model, model_params):
+        """ Initialize a IvimFit class instance.
+            Parameters
+            ----------
+            model : Model class
+            model_params : array
+                The parameters of the model. In this case it is an
+                array of ivim parameters. If the fitting is done
+                for multi_voxel data, the multi_voxel decorator will
+                run the fitting on all the voxels and model_params
+                will be an array of the dimensions (data[:-1], 4),
+                i.e., there will be 4 parameters for each of the voxels.
+        """
+        self.model = model
+        self.model_params = model_params
+
+    def __getitem__(self, index):
+        model_params = self.model_params
+        N = model_params.ndim
+        if type(index) is not tuple:
+            index = (index,)
+        elif len(index) >= model_params.ndim:
+            raise IndexError("IndexError: invalid index")
+        index = index + (slice(None),) * (N - len(index))
+        return type(self)(self.model, model_params[index])
+
+    @property
+    def S0_predicted(self):
+        return self.model_params[..., 0]
+
+    @property
+    def perfusion_fraction(self):
+        return self.model_params[..., 1]
+
+    @property
+    def D_star(self):
+        return self.model_params[..., 2]
+
+    @property
+    def D(self):
+        return self.model_params[..., 3]
+
+    @property
+    def shape(self):
+        return self.model_params.shape[:-1]
+
+    def predict(self, gtab, S0=1.):
+        """Given a model fit, predict the signal.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+               Gradient directions and bvalues
+
+        S0 : float
+            S0 value here is not necessary and will not be used to predict the
+            signal. It has been added to conform to the structure of the
+            predict method in multi_voxel which requires a keyword argument S0.
+
+        Returns
+        -------
+        signal : array
+            The signal values predicted for this model using its parameters.
+
+        """
+        return ivim_prediction(self.model_params, gtab)
+
+    def fill_na(self, bounds=((0., 0., 0., 0), (np.inf, 0.30, 0.01, 0.001)),
+                fill=(np.nan, np.nan, np.nan, np.nan)):
+        """
+        Function to fill nan values for parameters which are not within the bounds.
```

Comment:
The documentation here is a little confusing. At first I thought it was looking for any parameter but later it reads like only `f` is checked. But then looking at the code it's checking them all. Just some rewording here is all I'm thinking of.